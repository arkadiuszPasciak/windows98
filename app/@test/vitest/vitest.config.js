import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
