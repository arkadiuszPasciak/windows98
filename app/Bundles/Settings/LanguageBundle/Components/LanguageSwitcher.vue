<template>
  <UISelect
    id="languages"
    :model-value="i18n.locale.value"
    class="LanguageSwitcher"
    :options="MLanguageOptions"
    @update:model-value="
      ;(i18n.locale.value = $event),
        setLanguageVersionInStorage($event),
        updateLangAttribute($event)
    "
  />
</template>

<script setup lang="ts">
  import UISelect from '@APP|Bundles/UISelectBundle/Components/UISelect.vue'
  import { MLanguageOptions } from '@APP|Bundles/LanguageBundle/Mocks/Language.mocks'
  import {
    setLanguageVersionInStorage,
    updateLangAttribute,
  } from '@APP|Bundles/LanguageBundle/Services/Language.services'
  import { useI18n } from 'vue-i18n';

  const i18n = useI18n()
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/LanguageBundle/Styles/LanguageSwitcher.styles.scss"
/>
