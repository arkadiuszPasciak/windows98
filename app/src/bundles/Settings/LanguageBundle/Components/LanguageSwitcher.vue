<template>
  <template v-if="currentLanguage && languages">
    <UISelect
      id="languages"
      :model-value="currentLanguage"
      class="LanguageSwitcher"
      :options="languages"
      @update:model-value="toggleLanguage($event)"
    />
  </template>
</template>

<script setup lang="ts">
import UISelect from "@APP|Bundles/UISelectBundle/Components/UISelect.vue"
import { useLanguage } from "../hooks/use-language.hook"

const { currentLanguage, languages, toggleLanguage } = useLanguage()
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/LanguageBundle/Styles/LanguageSwitcher.styles.scss"
/>
