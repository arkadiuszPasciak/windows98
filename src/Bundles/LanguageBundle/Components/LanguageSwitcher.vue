<template>
  <select v-model="locale" class="LanguageSwitcher">
    <option
      v-for="language in languagesOptions"
      :key="`language-options-${language}`"
      :value="language"
    >
      {{ t(`Language.${language}`) }}
    </option>
  </select>
</template>

<script setup lang="ts">
  import { useI18n } from 'vue-i18n'
  import { languagesOptions } from '@Bundles/LanguageBundle/Mocks/Language.mocks'

  const { locale, t } = useI18n()
</script>

<i18n src="@Bundles/LanguageBundle/Locales/Language.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/LanguageBundle/Styles/LanguageSwitcher.styles.scss';
</style>
