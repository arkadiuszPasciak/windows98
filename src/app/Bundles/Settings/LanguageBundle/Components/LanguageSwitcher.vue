<template>
  <UISelect
    id="languages"
    :model-value="$i18n.locale"
    class="LanguageSwitcher"
    :options="MLanguageOptions"
    @update:model-value="
      ;($i18n.locale = $event),
        setLanguageVersionInStorage($event),
        updateLangAttribute($event)
    "
  />
</template>

<script setup lang="ts">
  import UISelect from '@Bundles/UISelectBundle/Components/UISelect.vue'
  import { MLanguageOptions } from '@Bundles/LanguageBundle/Mocks/Language.mocks'
  import {
    setLanguageVersionInStorage,
    updateLangAttribute,
  } from '@Bundles/LanguageBundle/Services/Language.services'
</script>

<i18n src="@Bundles/LanguageBundle/Locales/Language.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/LanguageBundle/Styles/LanguageSwitcher.styles.scss"
/>
