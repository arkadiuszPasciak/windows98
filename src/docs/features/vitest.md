# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
