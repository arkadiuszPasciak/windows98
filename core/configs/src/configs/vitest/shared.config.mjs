import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
