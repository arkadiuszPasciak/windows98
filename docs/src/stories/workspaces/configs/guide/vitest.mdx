import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

