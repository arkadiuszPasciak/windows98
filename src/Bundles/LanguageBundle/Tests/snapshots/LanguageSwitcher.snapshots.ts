import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import LanguageSwitcher from '@Bundles/LanguageBundle/Components/LanguageSwitcher.vue'

const defaultWrapper = mount(LanguageSwitcher)

describe('[LanguageBundle]<Snapshots>(LanguageSwitcher)', async () => {
  it('renders correctly component', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
