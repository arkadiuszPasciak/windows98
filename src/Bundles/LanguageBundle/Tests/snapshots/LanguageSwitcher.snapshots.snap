// Vitest Snapshot v1

exports[`[LanguageBundle]<Snapshots>(LanguageSwitcher) > renders correctly component 1`] = `
<select
  class="LanguageSwitcher"
  data-v-56cc1592=""
>
  
  <option
    data-v-56cc1592=""
    value="en"
  >
    English
  </option>
  <option
    data-v-56cc1592=""
    value="pl"
  >
    Polish
  </option>
  
</select>
`;
